pkg/asset/store.go

```diff
@@ -47,10 +108,19 @@ func (s *StoreImpl) fetch(asset Asset, indent string) error {
 		parents.Add(d)
 	}
 
-	logrus.Debugf("%sGenerating %s...", indent, asset.Name())
-	err := asset.Generate(parents)
+	// Before generating the asset, look if we have it all ready in the state file
+	// if yes, then use it instead
+	logrus.Debugf("%sFetching asset from state file: %s", indent, reflect.TypeOf(asset).String())
+	ok, err := s.GetStateAsset(asset)
 	if err != nil {
-		return errors.Wrapf(err, "failed to generate asset %s", asset.Name())
+		return errors.Wrapf(err, "failed to unmarshal asset from state file. Remove/fix the state file '%s' and continue..", stateFileName)
```

Comment:
We should never ask to user to fix the state file. 
this might be enough. `failed to unmashal assets {asset name} from state file{state file name}`